src/app/helpers.js

```diff
@@ -209,6 +209,37 @@ function capitalize(text) {
   return text.replace(/(?:^|\s)\S/g, a => a.toUpperCase());
 }
 
+/**
+* Get current project based on curent location and media.projects
+*/
+function getCurrentProject(projects) {
+  let project = null;
+  let currentProjectId = window.location.pathname.match(/project\/([0-9]+)/);
+  if (currentProjectId) {
+    currentProjectId = parseInt(currentProjectId[1], 10);
+    project = projects.edges.find(p => parseInt(p.node.dbid, 10) === currentProjectId);
+    if (project) {
+      project = project.node;
+    }
+  }
+  return project;
+}
+
+/**
+* Get current project id based on curent location and media.projects
+*/
+function getCurrentProjectId(projectIds) {
+  let projectId = null;
+  let currentProjectId = window.location.pathname.match(/project\/([0-9]+)/);
+  if (currentProjectId) {
+    currentProjectId = parseInt(currentProjectId[1], 10);
+    if (projectIds.indexOf(currentProjectId) > -1) {
+      projectId = currentProjectId;
+    }
+  }
+  return projectId;
+}
```

Comment:
I'm a bit confused here. Can't this just match and parseInt from the location and return? Perhaps name it `getCurrentProjectDbid` and call it from `getCurrentProject`?

Reply:
is it to avoid injecting any id in the URL, @melsawy ?

Reply:
yes Caio to make sure that id in URL belongs to project_ids list